File: Java_Generated/Java/RetrofitSample/Trading/app/src/main/java/tests/PiAct.java
package tests;

import android.util.Log;

import general_service.interfaces.IAction;
import generated.PI;
import measurements.Performer;
import measurements.interfaces.IDataConsumer;
import measurements.interfaces.IMeasurements;
import runtime.DataRuntimeConsumer;


public class PiAct extends PI implements IAction {
    public PiAct()
    {
        super();
        var c = getCategoryObjects();
        consumer =  performer.get(this, "Chart");
        measurements = consumer.getAllMeasurements();
    }

    @Override
    public void action() {
        for (var m : measurements) {
            var n = m.getMeasurementsCount();
            for (var i = 0; i < n; i++) {
                var o = m.getMeasurement(i);
                Log.d("", o + "\n");
            }
        }
    }

    IDataConsumer consumer;

    IMeasurements[] measurements;

    public void test()
    {
        var r = new DataRuntimeConsumer(consumer);
        var p = new Performer();
        p.performFixedStepCalculation(r, 0, 1, 1000, this);
    }

}

File: src/Web/reactprojectWatch/src/Tests/Wrappers/PIAct.ts
import { IDataRuntime } from "../../Library/Interfaces/IDataRuntime";
import { IDataConsumer } from "../../Library/Measurements/Interfaces/IDataConsumer";
import { PerformerMeasuremets } from "../../Library/Measurements/PerformerMeasuremets";
import { DataRuntimeConsumer } from "../../Library/Runtime/DataRuntimeConsumer";
import { PI } from "../PI";

export class PIAct extends PI {

    dc! : IDataConsumer;
    constructor() {
        super();
        var co = this.getCategoryObject("Chart");
        this.dc = co as unknown as IDataConsumer;
    }

    action(): void {
        var k = this.dc.getAllMeasurements()[0];
        var a = k.getMeasurement(0).getMeasurementValue();
  
        console.log(a);
    }

    func(): boolean {
        return false;
    }


    public test(): void {
        var runtime: IDataRuntime = new DataRuntimeConsumer(this.dc);
        var p: PerformerMeasuremets = new PerformerMeasuremets();
        p.performFixedStepCalculation(runtime, 0, 0.001, 1000, this, this);
    }


}